src/client/mod.rs

```diff
@@ -536,3 +559,100 @@ impl WeakClient {
         self.inner.upgrade().map(|inner| Client { inner })
     }
 }
+
+#[cfg(feature = "csfle")]
+pub struct EncryptedClientBuilder<const KMS_SET: bool> {
+    options: ClientOptions,
+    enc_options: AutoEncryptionOptions,
+}
+
+#[cfg(feature = "csfle")]
+impl<const KMS_SET: bool> EncryptedClientBuilder<KMS_SET> {
+    /// Add configuration for a single KMS provider.
```

Comment:
I think it would be good to include the note about already-added configurations in this docstring too, as well as an example showing that setting multiple providers is okay.

Reply:
Done.